queue_downloads.py

```diff
@@ -0,0 +1,301 @@
+        file_name (str): Name used for the Download request (with '_part_N').
+        email (str): Optional email to send notifications to.
+
+        input_file (str):
+            File containing newline delimited filepaths for the requested data.
+
+    Raises:
+        RuntimeError: If a status code other than 200 is returned.
+
+    Returns:
+        int: The Download id.
+    """
+    data = {
+        "sessionId": session_id,
+        "transport": transport,
+        "fileName": file_name,
+        "email": email,
+        "files": files,
+    }
+    response = requests.post(url=base_url + "/topcat/user/queue/files", data=data)
+    if response.status_code != 200:
+        raise RuntimeError(response.text)
+
+    content = json.loads(response.content)
+    download_id = content["downloadId"]
+    not_found = content["notFound"]
+    print(
+        f"Submitted part Download with id {download_id}\n"
+        f"{len(not_found)} file(s) could not be found:{not_found}\n"
+    )
+    return download_id
+
+
+def monitor(
+    base_url: str,
+    session_id: str,
+    downloads: "list[int]",
+    monitor_sleep: int,
+) -> None:
+    """
+    Checks the status of all `downloads` and prints this to the console. Will repeat
+    evert `monitor_sleep` seconds until all are complete.
+
+    Args:
+        session_id (str): ICAT session id.
+        downloads (list[int]): List of download ids for each part.
+        monitor_sleep (int): Number of seconds to wait between each check.
+
+    Raises:
+        RuntimeError: If a status code other than 200 is returned.
+    """
+    url = base_url + "/topcat/user/downloads/status"
+    params = {"sessionId": session_id, "downloadIds": downloads}
+    response = requests.get(url=url, params=params)
+    if response.status_code != 200:
+        raise RuntimeError(response.text)
+    content = json.loads(response.content)
+    print(content)
+
+    while any([s in {"QUEUED", "PAUSED", "PREPARING", "RESTORING"} for s in content]):
+        headers = {"Authorization": f"Bearer {session_id}"}
+        requests.put(url=base_url + "/datagateway-api/sessions", headers=headers)
+        if response.status_code != 200:
+            raise RuntimeError(response.text)
+
+        sleep(monitor_sleep)
+        response = requests.get(url=url, params=params)
+        if response.status_code != 200:
+            raise RuntimeError(response.text)
+
+        content = json.loads(response.content)
+        print(content)
+    
+    print("All downloads complete")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        prog="queue_downloads",
+        description=(
+            "Submits DataGateway Download requests for a list of specific filepaths. "
+            "The list will be split into separate parts of up to 10,000 files for "
+            "performance reasons. Once submitted Downloads will be visible in the "
+            "DataGateway UI as usual."
```

Comment:
Add something like "Alternatively use the monitor interval argument to allow the script to continue running and report on the progress of the download requests. Email notifications are another option."